src/groupeddataframe/groupeddataframe.jl

```diff
@@ -521,6 +521,117 @@ end
 
 Base.getproperty(key::GroupKey, p::AbstractString) = getproperty(key, Symbol(p))
 
+Base.hash(key::GroupKey, h::UInt) = _nt_like_hash(key, h)
+
+for fun in (Base.isequal, Base.:(==))
+    function $fun(k1::GroupKey, k2::GroupKey)
+        parent(k1).cols == parent(k2).cols || return false
+        return all(((a, b),) -> $fun(k1, k2), zip(k1, k2))
+    end
+
+    function $fun(k1::GroupKey, k2::DataFrameRow)
+        parent(k1).cols == _names(k2) || return false
+        return all(((a, b),) -> $fun(k1, k2), zip(k1, k2))
+    end
+
+    $fun(k1::DataFrameRow, k2::GroupKey) = $fun(k2, k1)
+
+    function $fun(k1::GroupKey, k2::NamedTuple)
+        parent(k1).cols == _propertynames(k2) || return false
+        return all(((a, b),) -> $fun(k1, k2), zip(k1, k2))
+    end
+
+    $fun(k1::NamedTuple, k2::GroupKey) = $fun(k2, k1)
+end
+
+for (fun1, fun2) in ((Base.isequal, Base.isless), (Base.:(==), Base.:(<)))
+
+    function $fun2(k1::GroupKey, k2::GroupKey)
+        length(k1) == length(k2) ||
+            throw(ArgumentError("compared GroupKeys must have the same number " *
+                                "of columns (got $(length(k1)) and $(length(k2)))"))
+        if parent(k1).cols != parent(k2).cols
+            mismatch = findfirst(i -> parent(k1).cols[i] != parent(k2).cols[i], 1:length(k1))
+            throw(ArgumentError("compared DataFrameRows must have the same column " *
+                                "names but they differ in column number $mismatch " *
+                                "where the names are :$(parent(k1).cols[mismatch]) and " *
+                                ":$(parent(k2).cols[mismatch]) respectively"))
+        end
+        for (a, b) in zip(k1, k2)
+            $fun1(a, b) || return $fun2(a, b)
+        end
+        return false
+    end
+
+    function $fun2(k1::GroupKey, k2::DataFrameRow)
+        length(k1) == length(k2) ||
+            throw(ArgumentError("compared objects must have the same number " *
+                                "of columns (got $(length(k1)) and $(length(k2)))"))
+        if parent(k1).cols != _names(k2)
```

Comment:
You defined a helper like this, the same definition would work for both types, right?
```julia
_getnames(x::GroupKey) = parent(x).cols 
_getnames(x::DataFrameRow) = _names(x)
```

Reply:
I will change this. I was thinking about it, but was too lazy ;(.